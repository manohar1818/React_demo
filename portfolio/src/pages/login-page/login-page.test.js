import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import LoginPage from './login-page';
  
const auth={
    login:jest.fn()
}
describe("LoginPage component",() =>{
    test("Matches the snapshot",() => {
        const input = create(< LoginPage auth={auth}/>);
        expect(input.toJSON()).toMatchSnapshot();

    })
    test("LoginPage element is rendered",() => {
        const { getByTestId, getByText } = render(<LoginPage auth={auth}/>);
        expect(getByTestId("loginPage")).toBeInTheDocument();
        
    });
   
   
   

});