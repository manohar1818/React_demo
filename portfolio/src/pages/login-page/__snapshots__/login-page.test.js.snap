// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component Matches the snapshot 1`] = `
<div
  data-testid="loginPage"
>
  <div
    data-testid="loginTemplate"
  >
     
    <div
      className="makeStyles-styleLogin-3"
      data-testid="login"
    >
      <h2
        className="MuiTypography-root makeStyles-input-2 MuiTypography-h2"
      >
        LoginPage
      </h2>
      <div
        className="makeStyles-forms-1"
      >
        <div
          className="makeStyles-button-4"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-testid="button"
            disabled={false}
            onBlur={[Function]}
            onClick={[MockFunction]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiTypography-root MuiTypography-button"
              >
                 
                login
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
